curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-66-diophantine-equation.md

```diff
@@ -28,37 +28,96 @@ By finding minimal solutions in x for D = {2, 3, 5, 6, 7}, we obtain the followi
 
 Hence, by considering minimal solutions in `x` for D ≤ 7, the largest `x` is obtained when D=5.
 
-Find the value of D ≤ 1000 in minimal solutions of `x` for which the largest value of `x` is obtained.
+Find the value of D ≤ `n` in minimal solutions of `x` for which the largest value of `x` is obtained.
 
 # --hints--
 
-`diophantineEquation()` should return a number.
+`diophantineEquation(7)` should return a number.
 
 ```js
-assert(typeof diophantineEquation() === 'number');
+assert(typeof diophantineEquation(7) === 'number');
 ```
 
-`diophantineEquation()` should return 661.
+`diophantineEquation(7)` should return `5`.
+
+```
+assert.strictEqual(diophantineEquation(7), 5);
+```
+
+`diophantineEquation(100)` should return `61`.
+
+```
+assert.strictEqual(diophantineEquation(100), 61);
+```
+
+`diophantineEquation(500)` should return `421`.
+
+```
+assert.strictEqual(diophantineEquation(500), 421);
+```
```

Comment:
````suggestion
`diophantineEquation(100)` should return `61`.

```
assert.strictEqual(diophantineEquation(100), 61);
```

`diophantineEquation(409)` should return `409`.

```
assert.strictEqual(diophantineEquation(409), 409);
```

`diophantineEquation(500)` should return `421`.

```
assert.strictEqual(diophantineEquation(500), 421);
```
````

This challenge presents the opportunity for an off-by-one error based on the camper's loop conditions. What are your thoughts on adding a test to ensure they are checking for `<=` `n`?

Reply:
That's a really good idea. It somehow didn't cross my mind to actively look for test cases like that, instead of just using rather arbitrary values...